### ISSUE TITLE
[World] Disable dynmap tracking outside of towns

### ISSUE THREAD

**Issue by alekb1** [EarthMC/Issue-Tracker#1398]
The dynmap is a great way of seeing the world. along with this, it is a powerful tool to plan routes, see where you can place towns, and who's out gathering sand, finding a spot to make a town, and mapping the area. With this immensely powerful tool, groups of 1-4 players will attack people in the wild. The best of these groups use invis pots so they are invisible on the map. The easy solution is that you yourself use invis pots, but that quickly gets very expensive. This is not about being against PvP, but rather that people should not be able to have a gps tracker on all the players in the wild.

My experience
I decided today that I wanted to continue of my mapping of Europe. After traveling for 1k blocks (5 min) I was attacked by the first player, which I escaped. after taking an hour break I continued and were attacked after walking for another 20 min by 4 players, which I were able to spot on the dynmap before they caught me. A couple of weeks ago I was making a map art, and were not as lucky, there I was jumped by 3 invis players while being thousands of blocks away from the nearest n spawn. They killed me and got my loot. These players would have never known that I were there if it had not been for the dynmap, and as such they had an unfair advantage over me.

In conclusion my suggestion is that you can still see people on the dynmap, but only when they are in towns. This is so that the map still gives the sensation that the server is bustling with life. If this is not possible I propose that players are not visible on the map at all unless they toggle it on. The fact that people can easily find everyone in the wild is simply to op, and it has a serious negative impact on the behavior of people in the wild, as people will be less willing to make map art, create infrastructure, and tour the server.

**Comment by Fruitloopins** [comment]
I think I would support this change or something similar to it if there were some other way to pvp on emc that didn't involve consent from the person you are attacking i.e. a war plugin or an alternative way to find where people are in wilderness that isn't so powerful. Without being able to see people on dynmap it would encourage people to use their heads to figure out where people are, for example if someone was frequently making map art and you could see them teleporting to a nation in Antarctica where they quickly go off map upon entering wilderness you would have to check each of the nearby map art canvases, this would also allow the player to attempt to trick players by walking in one direction out of claims and then switching their direction once they leave.

**Comment by alekb1** [comment]
Completely agree. You would still be able to stalk towns you are at war with, but you yourself would not have to down 4-5 invis pots while waiting for someone to leave. I'm very much for "smart" PvP in which players see patens, stalk towns, etc.. But currently it is just as simple as opening the dynmap, and finding the first person outside of a town.

**Comment by alekb1** [comment]
![1](https://user-images.githubusercontent.com/47500274/133658324-6f75c11c-6408-428a-84ec-64c90390fdd2.PNG)
An example of how bad it is. I was not invis for 30 seconds inside a town that I was not a part of. Truth be told there is no realistic part of being honed in on and attack if you leave your town, unless you are in an area with drones. Bandits hanging out in an area I can respect, but teleporting players who can track you and reach you in 5 min seems op.

**Comment by Goosius1** [comment]
FYI 
- In the [latest *SiegeWar* plugin release](https://github.com/TownyAdvanced/SiegeWar/releases/tag/0.5.2), just yesterday, many options were added for hiding players on the dynmap, including the suggested option of simply hiding them when in the wilderness.
- The hide feature can be used without any of the war features being enabled.

**Comment by alekb1** [comment]
That is great, this part could be great to add to EMC

**Comment by Basically-Andrew** [comment]
Instead of requiring SiegeWar,  [LlmDl's HuntPlayer](https://github.com/LlmDl/HuntPlayer) plugin could still allow players to be hunted. A message could also be sent to the hunted player when they are being hunted.
